dashboard/app/controllers/registrations_controller.rb

```diff
@@ -108,7 +109,9 @@ def finish_student_account
   # Get /users/new_sign_up/finish_teacher_account
   #
   def finish_teacher_account
-    @us_ip = us_ip?
+    location = Geocoder.search(request.ip).try(:first)
+    @country_code = location&.country_code.to_s.upcase
+    @us_ip = ['US', 'RD'].include?(@country_code)
```

Comment:
Wondering if it makes sense to only have `@country_code` passed in since `@us_ip` is based on it? Or do you think the frequent check to see if country code is `US` or `RD` wouldn't warrant it?

Reply:
Could even be a little helper function that checks it in `signUpFlowConstants.jsx` or something?

Reply:
Ugh I was grappling with this too (wrote a bit about it in the description). There's a helper on the back end (the us_ip? I deleted) but it's really just that one extra line. I agree that it seems weird to send both but I do think that doing it once here at least keeps it somewhat simple and then we don't have to refactor anything on the front end